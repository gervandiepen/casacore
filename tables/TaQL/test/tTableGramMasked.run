#!/bin/sh
#-----------------------------------------------------------------------------
# Script to test the masked array in TableGram and TableParse.
# All files generated will be deleted on exit.
#=============================================================================

# Create a symlink to achieve that casacore_memcheck (if used) will use
# that name 
rm -f tTableGramMasked
ln -s tTableGram tTableGramMasked

# Set default testsrcdir if undefined.
if test "$testsrcdir" = ""; then
  testsrcdir=../../../../../casacore/tables/TaQL/test
fi

# Use table tTable_2.data_v0 as the input by copying it.
rm -rf tTableGramMasked_tmp.tab*
mkdir tTableGramMasked_tmp.tab
cp -r $testsrcdir/../../Tables/test/tTable_2.data_v0/table.* tTableGramMasked_tmp.tab
chmod 644 tTableGramMasked_tmp.tab/*


$casa_checktool ./tTableGramMasked 'select sum(arr1),sum(arr1[arr1%3>0]),sum(arr1[arr1%3==0]) from tTableGramMasked_tmp.tab'

$casa_checktool ./tTableGramMasked 'select sum(arraydata(replacemasked(arr1[arr1%3>0], 1))) from tTableGramMasked_tmp.tab'

$casa_checktool ./tTableGramMasked 'select sum(arraydata(replaceunmasked(arr1[arr1%3>0], 1))) from tTableGramMasked_tmp.tab'

exit 0
