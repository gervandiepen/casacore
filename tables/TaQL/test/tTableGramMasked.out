select sum(arr1),sum(arr1[arr1%3>0]),sum(arr1[arr1%3==0]) from tTableGramMasked_tmp.tab
    has been executed
    select result of 10 rows
3 selected columns:  Col_1 Col_2 Col_3
 276 84 192
 852 276 576
 1428 468 960
 2004 660 1344
 2580 852 1728
 3156 1044 2112
 3732 1236 2496
 4308 1428 2880
 4884 1620 3264
 5460 1812 3648
select sum(arraydata(replacemasked(arr1[arr1%3>0], 1))) from tTableGramMasked_tmp.tab
    has been executed
    select result of 10 rows
1 selected columns:  Col_1
 100
 292
 484
 676
 868
 1060
 1252
 1444
 1636
 1828
select sum(arraydata(replaceunmasked(arr1[arr1%3>0], 1))) from tTableGramMasked_tmp.tab
    has been executed
    select result of 10 rows
1 selected columns:  Col_1
 200
 584
 968
 1352
 1736
 2120
 2504
 2888
 3272
 3656
